bson/src/main/org/bson/codecs/pojo/PojoCodecProvider.java

```diff
@@ -72,6 +74,17 @@ public static Builder builder() {
                 classModel = createClassModel(clazz, conventions);
             }
             return new PojoCodecImpl<T>(classModel, registry, discriminatorLookup);
+        } else if (automatic) {
+            try {
+                classModel = createClassModel(clazz, conventions);
+            } catch (IllegalStateException e) {
+                return null;
+            }
+            if (!clazz.isInterface() && classModel.getPropertyModels().isEmpty()) {
```

Comment:
Because interfaces can have no properties, we sometimes need to have a class model for them. They do rely on the discriminator to work to rehydrate, which the PojoCodec will handle.